gapir/cc/resource_cache.cpp

```diff
@@ -23,51 +23,105 @@
 #include <vector>
 
 namespace gapir {
-size_t ResourceCache::prefetch(const Resource* res, size_t count,
-                               ResourceLoader* fetcher) {
-  size_t res_sum = 0;
-  std::vector<Resource> uncached;
-  uncached.reserve(count);
-  size_t alreadyCached = 0;
+
+void ResourceCache::setPrefetch(const Resource* resources, size_t count,
+                                std::unique_ptr<ResourceLoader> fetcher) {
+  mResources.clear();
+  mResourceIterators.clear();
+
+  mResources.reserve(count);
+
+  for (unsigned int i = 0; i < count; ++i) {
+    mResources.push_back(resources[i]);
+    mResourceIterators[resources[i].getID()] = mResources.end() - 1;
+  }
+
+  mFetcher = std::move(fetcher);
+}
+
+std::vector<Resource> ResourceCache::anticipateNextResources(
+    const Resource& resource, size_t bytesToFetch) {
+  std::vector<Resource> expectedResources;
+
+  size_t bytesSoFar = 0;
+
+  auto resMapIter = mResourceIterators.find(resource.getID());
+  if (resMapIter == mResourceIterators.end()) {
+    return expectedResources;  // We don't know about this resource so we're
+                               // blind. Return the empty vector.
+  }
+
+  std::vector<Resource>::iterator resIter = resMapIter->second;
```

Comment:
I am fine either way, but we often use auto for these. I prefer NOT using auto, but iterators get annoying to type.

Reply:
Changed to auto. I have mixed feelings on auto and tend to use it by gut feeling. Still trying to understand the rules behind those gut feels. I agree that iterators are strong contenders for the use of auto though!